src/core/ctxt.ml

```diff
@@ -1,17 +1,25 @@
 (** Typing context. *)
 
 open Term
+open Lplib
 open Timed
 
 (** [unbind ctx a def b] returns a triple [(x,t,new_ctx)] such that [(x,t)] is
-   an unbinding of [b] (in the sense of [Bindlib.unbind]) and [new_ctx] is an
-   extension of the context [ctx] with the declaration that [x] has type [a]
-   (only if [x] occurs in [t]). If [def] is of the form [Some(u)], the context
-   also registers the term [u] as the definition of variable [x]. *)
+    an unbinding of [b] (in the sense of [Bindlib.unbind]) and [new_ctx] is an
+    extension of the context [ctx] with the declaration that [x] has type [a].
+    If [def] is of the form [Some(u)], the context also registers the term [u]
+    as the definition of variable [x]. *)
 let unbind : ctxt -> term -> term option -> tbinder -> tvar * term * ctxt =
   fun ctx a def b ->
   let (x, t) = Bindlib.unbind b in
-  (x, t, if Bindlib.binder_occur b then (x, a, def) :: ctx else ctx)
+  (x, t, (x, a, def) :: ctx)
```

Comment:
Why do you want to extend the context if the variable is not used?
If you do that, then this function becomes useless and you can unfold it.